tests/functional/detection_status/test_detection_status.py

```diff
@@ -0,0 +1,159 @@
+#
+# -----------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -----------------------------------------------------------------------------
+
+""" detection_status function test.
+
+WARNING!!!
+This is a generated test skeleton remove the parts not required by the tests.
+WARNING!!!
+
+"""
+import json
+import os
+import unittest
+
+import shared
+
+from libtest import codechecker
+from libtest import env
+
+
+class TestSkeleton(unittest.TestCase):
+
+    def setUp(self):
+        """
+        WARNING!!!
+        This is an example how to get the configurations needed by the tests.
+        WARNING!!!
+        """
+
+        # TEST_WORKSPACE is automatically set by test package __init__.py .
+        self.test_workspace = os.environ['TEST_WORKSPACE']
+
+        test_class = self.__class__.__name__
+        print('Running ' + test_class + ' tests in ' + self.test_workspace)
+
+        self._codechecker_cfg = env.import_codechecker_cfg(self.test_workspace)
+
+        # Get the CodeChecker cmd if needed for the tests.
+        self._codechecker_cmd = env.codechecker_cmd()
+        self._test_dir = os.path.join(os.path.dirname(__file__), 'test_files')
+
+        try:
+            os.makedirs(self._test_dir)
+        except os.error:
+            # Directory already exists.
+            pass
+
+        # Setup a viewer client to test viewer API calls.
+        self._cc_client = env.setup_viewer_client(self.test_workspace)
+        self.assertIsNotNone(self._cc_client)
+
+        # Change working dir to testfile dir so CodeChecker can be run easily.
+        os.chdir(self._test_dir)
+
+        self._source_file = "main.cpp"
+
+        # Init project dir.
+        project_info = {
+            "name": "hello",
+            "clean_cmd": "",
+            "build_cmd": "g++ main.cpp"
+        }
+
+        with open(os.path.join(self._test_dir, 'project_info.json'), 'w') as f:
+            json.dump(project_info, f)
+
+    def _create_source_file(self, version):
+        if version == 1:
+            source = """
```

Comment:
Perhaps we could put the later hardcoded bug hashes here at least as a comment to see which means which bug in the code.